src/anomalib/deploy/model.py

```diff
@@ -0,0 +1,277 @@
+        image = image.div(255.0)
+        processed_image = self.transform(image)
+
+        # if len(processed_image) == 3:
+        #     processed_image = processed_image.unsqueeze(0)
+
+        return processed_image
+
+    def forward(self, image: torch.Tensor) -> Result:
+        """Forward pass of the model.
+
+        Args:
+            image (torch.Tensor): Input tensor.
+
+        Returns:
+            torch.Tensor: Output tensor.
+        """
+        image = self.pre_process(image)
+        outputs = self.model(image)
+        return self.post_process(predictions=outputs, metadata=self.metadata)
+
+    def batch_mask_to_boxes(self, pred_mask: Tensor):
+        """Converts a batch of masks to bounding boxes.
+
+        Args:
+            pred_mask (Tensor): Batch of masks
+
+        Returns:
+            Tensor: Bounding boxes
+            Tensor: Boxes per image. Since the bounding boxes are of shape N x 4, this informs how many boxes are
+                present in each image.
+        """
+        height, width = pred_mask.shape[-2:]
+        pred_mask = pred_mask.view((-1, 1, height, width)).float()  # reshape to (B, 1, H, W) and cast to float
+
+        batch_comps = connected_components(pred_mask, num_iterations=1000).squeeze(1).int()
+        B, H, W = batch_comps.shape
+
+        # split the boxes later on based on this
+        # since torch.count_nonzero is not supported in onnx, we use torch.sum with non equality
+        boxes_per_image = torch.sum(torch.unique(batch_comps.view(B, -1), dim=1) != 0, dim=1) - 1
+
+        labels = torch.unique(batch_comps)[1:]
+        masks = torch.ones((B, labels.size(0), H, W), device=pred_mask.device, dtype=torch.int64)
+        # multipy labels to masks along channel dimension to get each category in a different channel
+        masks = masks * labels.unsqueeze(1).unsqueeze(2)
+        masks = masks == batch_comps.unsqueeze(1)  # B x N x H x W
+
+        bboxes = torch.full(
+            (labels.shape[0], 4), fill_value=-1.0, device=pred_mask.device, dtype=torch.int64
+        )  # N-1 x 4, since lables are unique and 0 is background
+
+        # since the mask categories are mutually exclusive across batch, we can sum them to remove the batch dimension
+        masks = torch.sum(masks, dim=0)  # N-1 x H x W
+        # Get the max and min values of the masks by multiplying each mask by a grid of its coordinates
+        masks_ = masks.unsqueeze(1) * torch.stack(torch.meshgrid(torch.arange(H), torch.arange(W)), dim=0).to(
+            pred_mask.device
+        )  # N-1 x 2 x H x W
+        masks_ = masks_.view(labels.size(0), 2, -1)  # N-1 x 2 x H*W
+        # # assign bounding boxes
+        bboxes[:, 2] = torch.max(masks_[:, 1], dim=1)[0]
+        bboxes[:, 3] = torch.max(masks_[:, 0], dim=1)[0]
+        masks_[masks_ == 0] = H + 1
+        bboxes[:, 0] = torch.min(masks_[:, 1], dim=1)[0]
+        bboxes[:, 1] = torch.min(masks_[:, 0], dim=1)[0]
+        return bboxes, boxes_per_image
```

Comment:
instead of returning `boxes_per_image`, maybe you could prepend the image idx to the box coordinates, like
```python
indices = torch.repeat_interleave(torch.arange(B), boxes_per_image)
torch.cat([indices.unsqueeze(1), bboxes], dim=1)
```
This leads to better interpretable outputs and might make further processing easier.
(Just a suggestion, your call)

Reply:
I get this error when I try to use `repeat_interleave`
```bash
TypeError: z_(): incompatible function arguments. The following argument types are supported:
    1. (self: torch._C.Node, arg0: str, arg1: torch.Tensor) -> torch._C.Node

Invoked with: %25415 : Tensor = onnx::Constant(), scope: __main__.Dummy::
```
I think this is because the inputs to `repeat_interleave` have a variable dimension.

Reply:
That's weird, `repeat_interleave` is also used in RKDE implementation, which does export to onnx without problems

Reply:
When I explicitly add the dimenstion `indices.repeat_interleave(boxes_per_image, 0)` onnx export works but OpenVINO IR conversion fails.
```bash
RuntimeError: Check 'unknown_operators.empty()' failed at src/frontends/onnx/frontend/src/core/graph.cpp:213:
OpenVINO does not support the following ONNX operations: ConcatFromSequence, SequenceEmpty, SplitToSequence
```
But looks like the support is in progress https://github.com/openvinotoolkit/openvino/issues/15094

Reply:
Again, OpenVINO conversion works fine for RKDE. So it's a bit surprising that it doesn't work here.